source/common/json/json_loader.h

```diff
@@ -20,6 +27,186 @@ class Factory {
   static ObjectPtr loadFromString(const std::string& json);
 
   static const std::string listAsJsonString(const std::list<std::string>& items);
+
+private:
+  /*
+   * Internal representation of Object.
+   */
+  class Field;
+  typedef std::shared_ptr<Field> FieldPtr;
+
+  class Field : public Object, public std::enable_shared_from_this<Field> {
```

Comment:
ah cool was not aware of this feature. new to the language. will make that change.